DFC.ServiceTaxonomy.JobProfiles.Module/ServiceBusHandling/DataEventProcessor.cs

```diff
@@ -114,17 +119,41 @@ private async Task GenerateServiceBusMessageForJobProfile(ContentContextBase con
             await _serviceBusMessageProcessor.SendJobProfileMessage(jobprofileMessage, context.ContentItem.ContentType, actionType);
         }
 
-        private Task GenerateServiceBusMessageForWYDTypes(ContentContextBase context, string actionType)
+        private async Task GenerateServiceBusMessageForWYDTypes(ContentContextBase context, string actionType)
         {
-            // TODO: find all parents for the referenced content item.
-            IEnumerable<WhatYouWillDoContentItem> jobprofileData = new List<WhatYouWillDoContentItem>
+            bool isSaved = actionType.Equals(ActionTypes.Published) || actionType.Equals(ActionTypes.Draft);
+            IList<WhatYouWillDoContentItem> jobprofileData = new List<WhatYouWillDoContentItem>();
+            string FieldDescription = context.ContentItem.ContentType switch
+            {
+                ContentTypes.Environment => context.ContentItem.Content.Environment.Description.Html,
+                ContentTypes.Uniform => context.ContentItem.Content.Uniform.Description.Html,
+                ContentTypes.Location => context.ContentItem.Content.Location.Description.Html,
+                _ => throw new ArgumentException("No valid match found"),
+            };
+
+            var matches = await _jobProfileIndexRepository.GetAll(b => b.Uniform != null && b.Uniform.Contains(context.ContentItem.ContentItemId) ||
+                b.Environment != null && b.Environment.Contains(context.ContentItem.ContentItemId) ||
+                b.Location != null && b.Location.Contains(context.ContentItem.ContentItemId)).ListAsync();
+
+            foreach (var item in matches)
+            {
+                if (isSaved)
                 {
-                    new WhatYouWillDoContentItem()
-                    {
-                        Id = context.ContentItem.As<GraphSyncPart>().ExtractGuid(),
-                        Title = context.ContentItem.Content.Title}
-                };
-            return _serviceBusMessageProcessor.SendOtherRelatedTypeMessages(jobprofileData, context.ContentItem.ContentType, actionType);
+                    jobprofileData.Add(new WhatYouWillDoContentItem() { Id = context.ContentItem.As<GraphSyncPart>().ExtractGuid(), Title = context.ContentItem.Content.TitlePart.Title,
```

Comment:
The formatting is a bit off making it difficult to read, please format the document (Hint: Ctrl+K, Ctrl+D) :)